### ISSUE TITLE
gpu-operator v1.5.2 on OCP 4.5.31 Readiness probe failed: stat: cannot stat '/tmp/operator-sd-ready': No such file or directory

### ISSUE THREAD

**Issue by worsco** [NVIDIA/gpu-operator#148]
### 1. Issue or feature description

Cloud Provider is AWS, platform is OCP 4.5.31. Installed gpu-operator 1.5.1 and works fine.  v1.5.2 became available in OperatorHub, Upgraded to gpu-operator 1.5.2 fails with Readiness probe failed: stat: cannot stat '/tmp/operator-sdk-ready': No such file or directory

### 2. Steps to reproduce the issue

Succesfully install gpu-operator 1.5.1 and nfd.  Test by instantiating an AWS EC2 with GPU and validate that nvidia-smi works.

Scale machinesets with GPUs EC2s to zero.

Upgrade to gpu-operator 1.5.2 via OperatorHub upgrade.  gpu-operator pod never becomes ready.  Constantly fails with: Readiness probe failed: stat: cannot stat '/tmp/operator-sdk-ready': No such file or directory

### 3. Information to [attach](https://help.github.com/articles/file-attachments-on-issues-and-pull-requests/) (optional if deemed irrelevant)

attaching logs from operational v1.5.1 gpu-operator pod and non-operational v1.5.2 pod.

Note that there are no nodes at this time with GPUs (scaled GPU nodes to zero replicas when not running GPU loads or testing).

[gpu-operator-5f459b7f9f-7z7zl-gpu-operator-v1.5.1.log](https://github.com/NVIDIA/gpu-operator/files/5978591/gpu-operator-5f459b7f9f-7z7zl-gpu-operator-v1.5.1.log)
[gpu-operator-c559f4bf8-hbvcp-gpu-operator-v1.5.2.log](https://github.com/NVIDIA/gpu-operator/files/5978592/gpu-operator-c559f4bf8-hbvcp-gpu-operator-v1.5.2.log)

**Comment by shivamerla** [comment]
@worsco logs indicate that multiple instances of gpu-operator pods are running hence newer version pod cannot become leader. `/tmp/operator-sdk-ready` file is created soon after leader-election is done. Can you un-install operator, ensure gpu-operator deployment/pods are deleted from `openshift-operators` project and try re-install again?

```
{"level":"info","ts":1613338875.482324,"logger":"leader","msg":"Found existing lock","LockOwner":"gpu-operator-5f459b7f9f-7pl5p"}
{"level":"info","ts":1613338875.4942334,"logger":"leader","msg":"Not the leader. Waiting."}
{"level":"info","ts":1613338876.6242375,"logger":"leader","msg":"Not the leader. Waiting."}
{"level":"info","ts":1613338879.0132816,"logger":"leader","msg":"Not the leader. Waiting."}
{"level":"info","ts":1613338883.5577,"logger":"leader","msg":"Not the leader. Waiting."}
{"level":"info","ts":1613338892.2717235,"logger":"leader","msg":"Not the leader. Waiting."}
{"level":"info","ts":1613338909.642118,"logger":"leader","msg":"Not the leader. Waiting."}
{"level":"info","ts":1613338927.8510668,"logger":"leader","msg":"Not the leader. Waiting."}
{"level":"info","ts":1613338944.0737321,"logger":"leader","msg":"Not the leader. Waiting."}
{"level":"info","ts":1613338960.5902593,"logger":"leader","msg":"Not the leader. Waiting."}
{"level":"info","ts":1613338976.9148564,"logger":"leader","msg":"Leader pod has been deleted, waiting for garbage collection do remove the lock."}
{"level":"info","ts":1613338993.8924198,"logger":"leader","msg":"Leader pod has been deleted, waiting for garbage collection do remove the lock."}
{"level":"info","ts":1613339011.5593414,"logger":"leader","msg":"Leader pod has been deleted, waiting for garbage collection do remove the lock."}
{"level":"info","ts":1613339030.1791008,"logger":"leader","msg":"Leader pod has been deleted, waiting for garbage collection do remove the lock."}
{"level":"info","ts":1613339046.8760984,"logger":"leader","msg":"Leader pod has been deleted, waiting for garbage collection do remove the lock."}
```

**Comment by worsco** [comment]
Is the workflow of the gpu-operator (assume nfd is already installed)?

1. Install the gpu-operator v1.5.1 from OperatorHub and set to automatic
2. Apply the ClusterPolicy
3. A new version of gpu-operator v1.5.2 is available on OperatorHub
4. Because it is set to automatic, it will fail
5. Uninstall gpu-operator
6. Install newest gpu-operator
7. Apply the new ClusterPolicy with new container image definitions

When the subscription for gpu-operator is set to automatic (instead of manual), it will attempt to upgrade the older, running gpu-operator to the newer version.   Should the instructions state that we should be setting the subscription to manual?

**Comment by shivamerla** [comment]
@worsco yes, will update documentation to set subscription to manual approval. We want users to manually delete the old version/clusterpolicy instance and install new operator/clusterpolicy instance. Currently updates to clusterpolicy are not handled and we plan to add that soon. One of the major reason being drivers once loaded cannot be automatically upgraded(unloaded, recompiled and installed again) without draining workloads first. But other components can be upgraded via ClusterPolicy update.

**Comment by shivamerla** [comment]
@worsco We have released `v1.7.0` of operator which uses latest operator-sdk `v1.4` and readiness probes are moved to `http` based. Also, new channel `v1.7` created and subscription needs to be created with this channel. Existing installed versions have to be uninstalled and new subscription/operator has to be installed.
